Source/JavaScriptCore/parser/ASTBuilder.h

```diff
@@ -281,7 +281,8 @@ class ASTBuilder {
     {
         if (base->isSuperNode())
             usesSuperProperty();
-        
+        if (base->isArguments(m_vm) && *property == m_vm.propertyNames->length)
+            m_scope.m_usesArgumentsLength = true;
```

Comment:
I don't think we should do this. Remove it.
What we should do is marking `arguments` resolve by default, and not marking usesArguments only when we saw `arguments.length`. The current approach is very fragile and dangerous.